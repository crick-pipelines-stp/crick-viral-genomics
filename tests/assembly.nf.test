nextflow_pipeline {
    name "Test Pipeline Components"
    script "../main.nf"

    tag "assembly"

    test("assembly_illumina") {
        tag "assembly/illumina"
        when {
            params {
                run_assemble_ref = true
                run_reporting    = false

                outdir      = "$outputDir"
                viral_fasta = "$projectDir/tests/data/seq_sim/influenza_A/*"
                samplesheet = "$projectDir/tests/data/samplesheets/test_illumina_flu_amp_small.csv"
            }
        }
        then {
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.trace.succeeded().size(),
                    stable_name,
                    stable_path
                ).match() }
            )
        }
    }
}
