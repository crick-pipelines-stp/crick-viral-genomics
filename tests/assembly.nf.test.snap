{
    "assembly_illumina": {
        "content": [
            23,
            [
                "consensus",
                "consensus/consensus.merged.fasta",
                "pipeline_info",
                "ref_assembly",
                "ref_assembly/2-A_SouthSudan_1029_2024.blast.txt",
                "ref_assembly/2-A_SouthSudan_1029_2024.filtered.fasta",
                "ref_assembly/2-A_SouthSudan_1029_2024.ref_segs.txt",
                "ref_assembly/2-A_SouthSudan_1029_2024.tophits.fasta",
                "ref_assembly/2-A_SouthSudan_1029_2024.tophits.fasta.fai",
                "ref_assembly/20-A_Tajikistan_12-928_2023.blast.txt",
                "ref_assembly/20-A_Tajikistan_12-928_2023.filtered.fasta",
                "ref_assembly/20-A_Tajikistan_12-928_2023.ref_segs.txt",
                "ref_assembly/20-A_Tajikistan_12-928_2023.tophits.fasta",
                "ref_assembly/20-A_Tajikistan_12-928_2023.tophits.fasta.fai",
                "ref_assembly/51-A_Spain_1810_2024_S51.blast.txt",
                "ref_assembly/51-A_Spain_1810_2024_S51.filtered.fasta",
                "ref_assembly/51-A_Spain_1810_2024_S51.ref_segs.txt",
                "ref_assembly/51-A_Spain_1810_2024_S51.tophits.fasta",
                "ref_assembly/51-A_Spain_1810_2024_S51.tophits.fasta.fai",
                "ref_assembly/cd_hit",
                "ref_assembly/cd_hit/2-A_SouthSudan_1029_2024.fasta",
                "ref_assembly/cd_hit/2-A_SouthSudan_1029_2024.fasta.clstr",
                "ref_assembly/cd_hit/20-A_Tajikistan_12-928_2023.fasta",
                "ref_assembly/cd_hit/20-A_Tajikistan_12-928_2023.fasta.clstr",
                "ref_assembly/cd_hit/51-A_Spain_1810_2024_S51.fasta",
                "ref_assembly/cd_hit/51-A_Spain_1810_2024_S51.fasta.clstr",
                "ref_assembly/spades",
                "ref_assembly/spades/2-A_SouthSudan_1029_2024",
                "ref_assembly/spades/2-A_SouthSudan_1029_2024/spades",
                "ref_assembly/spades/2-A_SouthSudan_1029_2024/spades/contigs.fasta",
                "ref_assembly/spades/20-A_Tajikistan_12-928_2023",
                "ref_assembly/spades/20-A_Tajikistan_12-928_2023/spades",
                "ref_assembly/spades/20-A_Tajikistan_12-928_2023/spades/contigs.fasta",
                "ref_assembly/spades/51-A_Spain_1810_2024_S51",
                "ref_assembly/spades/51-A_Spain_1810_2024_S51/spades",
                "ref_assembly/spades/51-A_Spain_1810_2024_S51/spades/contigs.fasta"
            ],
            [
                "consensus.merged.fasta:md5,d41d8cd98f00b204e9800998ecf8427e",
                "2-A_SouthSudan_1029_2024.blast.txt:md5,f17f4d168ff778b3e7f7a39c1c792912",
                "2-A_SouthSudan_1029_2024.filtered.fasta:md5,d77d4388ce3ebd4a0dcd10e7b27b646d",
                "2-A_SouthSudan_1029_2024.ref_segs.txt:md5,00978ec9f2836c42b3ce4cf81a607357",
                "2-A_SouthSudan_1029_2024.tophits.fasta:md5,31297e49ebd74f2b22c111f02bbe8592",
                "2-A_SouthSudan_1029_2024.tophits.fasta.fai:md5,c201df12e2571e134bbe6015c3cc0412",
                "20-A_Tajikistan_12-928_2023.blast.txt:md5,9081f4cbdd6c63f4f117bf67a6ca3c67",
                "20-A_Tajikistan_12-928_2023.filtered.fasta:md5,bede7f8dd3363c6ae943f84ff04e7255",
                "20-A_Tajikistan_12-928_2023.ref_segs.txt:md5,a7b1a9ebc230996feeb7dceacc925ecf",
                "20-A_Tajikistan_12-928_2023.tophits.fasta:md5,4bd73fb691713eb2a9bb60f7728fe9c6",
                "20-A_Tajikistan_12-928_2023.tophits.fasta.fai:md5,3fb5fe22da13e7903d208038f9b5a52c",
                "51-A_Spain_1810_2024_S51.blast.txt:md5,652caaab3dc77cdddba3f4b05425cf15",
                "51-A_Spain_1810_2024_S51.filtered.fasta:md5,7a80150d4785cd6c60b07055886b40a2",
                "51-A_Spain_1810_2024_S51.ref_segs.txt:md5,d8870afcfcbc44c5b0aa981849d0c461",
                "51-A_Spain_1810_2024_S51.tophits.fasta:md5,4f6fc8921940abe2780311d4ed7f9d6f",
                "51-A_Spain_1810_2024_S51.tophits.fasta.fai:md5,cba9b4ace478fa5bc73284933526763c",
                "2-A_SouthSudan_1029_2024.fasta:md5,f837b5c4b523d5f606318a1ba6e02a08",
                "2-A_SouthSudan_1029_2024.fasta.clstr:md5,81569cde00622a33d2af56073512cfe8",
                "20-A_Tajikistan_12-928_2023.fasta:md5,6e9f2453a3d6f252a37cb38e5153b390",
                "20-A_Tajikistan_12-928_2023.fasta.clstr:md5,5bcc904ac5c3672f0f660f574cd903bc",
                "51-A_Spain_1810_2024_S51.fasta:md5,91857b4f86680a4e8871e74d0dafc4b3",
                "51-A_Spain_1810_2024_S51.fasta.clstr:md5,59b668fec1082891e3d992b8c1a423d7",
                "contigs.fasta:md5,3e78445d059fd3a4b78b870982ecd265",
                "contigs.fasta:md5,655039f9330ad5683de1a677147379a0",
                "contigs.fasta:md5,049dc385d6f559981ab71fabac2f5fed"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-07T11:54:28.492132"
    }
}